exist-start/src/main/java/org/exist/start/CompatibleJavaVersionCheck.java

```diff
@@ -24,37 +24,49 @@
 import java.util.Optional;
 import java.util.stream.Stream;
 
-import static org.exist.start.CompatibleJavaVersionCheck.IncompatibleJavaVersion.IncompatibleJavaVersion;
+import static org.exist.start.CompatibleJavaVersionCheck.IncompatibleJavaVersion.create;
 import static org.exist.start.Main.ERROR_CODE_INCOMPATIBLE_JAVA_DETECTED;
 
+/**
+ * Helper class for checking OpenJDK compatibility.
+ * <p>
+ * eXist-db has been compiled with Java21 (state of 2025Q2).
+ * <p>
+ * Older versions of Java contained a number of serious compiler bugs that caused database corruptions.
+ * These problematic versions are deprecated and therefore this class is not relevant anymore.
+ * <p>
+ * The code is kept for archival purposes and potential future re-usage.
+ * <p>
+ * ----------------------------
+ * <p>
+ * OpenJDK versions 12 through 15.0.1 suffer from a critical bug in the JIT C2 compiler that will
+ * cause data loss in eXist-db. The problem has been reported to the OpenJDK community.
+ * <p>
+ * For more information, see:
+ * - <a href="https://bugs.openjdk.java.net/browse/JDK-8253191">C2: Masked byte comparisons with large masks produce wrong result on x86</a>
+ * - <a href="https://github.com/eXist-db/exist/issues/3375">eXist-db does not run correctly on JDKs 12, 13, 14 and 15 #3375</a>
+ *
+ *
+ */
 public class CompatibleJavaVersionCheck {
 
     private static final IncompatibleJavaVersion[] INCOMPATIBLE_JAVA_VERSIONS = {
-            IncompatibleJavaVersion(12),
-            IncompatibleJavaVersion(13),
-            IncompatibleJavaVersion(14),
-            IncompatibleJavaVersion(15, 0, 2)
+            create(12),
```

Comment:
If we decide to compile start.jar with java21 iso java8 .... these items could be removed; voiding all tests too,